org.eclipse.jdt.ls.debug/src/org/eclipse/jdt/ls/debug/adapter/variables/JdiObjectProxy.java

```diff
@@ -0,0 +1,58 @@
+/*******************************************************************************
+ * Copyright (c) 2017 Microsoft Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     Microsoft Corporation - initial API and implementation
+ *******************************************************************************/
+
+package org.eclipse.jdt.ls.debug.adapter.variables;
+
+import java.util.Objects;
+
+public class JdiObjectProxy<T> {
```

Comment:
the jdi impl calls the valid state function in hashcode, if we resume a thread, the related stackframe will throw invalid state exception when evaluating its hashcode, but we need the hashcode of the stackframe to remove the stackframe, so I create a wrapper object:
    private final Map<V, Set<O>> referenceMap = new HashMap<>();
    private final Map<V, Integer> objectIdMap = new HashMap<>();
...
public boolean removeObjectById(int id) {
        synchronized (this) {
            V object = this.objectCollection.remove(id);
            if (object == null)  {
                return false;
            }
            referenceMap.remove(object);
            objectIdMap.remove(object);
            return true;
        }
    }